kani-compiler/src/kani_middle/resolve.rs

```diff
@@ -0,0 +1,236 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! This module contains code for resolving strings to `DefId`s.
```

Comment:
Can you please expand this description? What kind of strings are we talking about? Does this module cover any string that represents a path? Do we support fully qualified paths? Maybe adding a link to the path syntax might help too: https://doc.rust-lang.org/reference/paths.html.

Which brings me a question, do we support qualified paths?

Reply:
Looking through the open issues kind of answers my last question about qualified paths: https://github.com/model-checking/kani/issues/1997

But I think it would be great if we can explicitly state that we are only handling simple paths and add a link to its definition: https://doc.rust-lang.org/reference/paths.html#simple-paths

Reply:
I've updated some comments to make this clearer (and included the link for simple paths)